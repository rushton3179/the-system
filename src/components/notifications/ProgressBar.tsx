import { motion } from 'framer-motion';
import React, { FC } from 'react';
import styles from './ProgressBar.module.scss';

interface Props {
  duration: number;
  onCompletion?(): void;
}

const ProgressBar: FC<Props> = ({ duration, onCompletion }) => {
  return (
    <div className={styles.container}>
      <motion.div
        className={styles.progress}
        initial={{ width: '0%' }}
        animate={{ width: '100%', transition: { duration } }}
        onAnimationComplete={onCompletion}
      />
    </div>
  );
};

export default ProgressBar;
