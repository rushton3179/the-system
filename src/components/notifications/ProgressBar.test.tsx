import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import ProgressBar from './ProgressBar';

describe('NavMenu', () => {
  afterEach(cleanup);

  test('it renders', () => {
    render(<ProgressBar duration={0} />);

    expect(screen.getByTestId('progress-bar')).toBeInTheDocument();
  });

  test('it matches snapshot', () => {
    const { container } = render(<ProgressBar duration={0} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('it renders nothing when no duration', () => {
    render(<ProgressBar />);

    expect(screen.queryByTestId('progress-bar')).not.toBeInTheDocument();
  });
});
