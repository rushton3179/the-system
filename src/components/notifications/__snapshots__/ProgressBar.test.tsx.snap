// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu it matches snapshot 1`] = `
<div
  class="container"
  data-testid="progress-bar"
>
  <div
    class="progress"
    style="width: 0%;"
  />
</div>
`;
